src/typechecker/tc-env.lisp

```diff
@@ -47,57 +48,58 @@
            (values tc:ty tc:ty-predicate-list))
 
-
-  (let* ((scheme (or (gethash (parser:node-variable-name var) (tc-env-ty-table env))
-
-                     (tc:lookup-value-type (tc-env-env env) (parser:node-variable-name var) :no-error t)
-
-                     ;; Binding is unknown. Create an error.
-                     (let* ((sym-name (symbol-name (parser:node-variable-name var)))
-                            (matches (append
-                                      (remove-if-not
-                                       (lambda (s) (string= (symbol-name s) sym-name))
-                                       (alexandria:hash-table-keys (tc-env-ty-table env)))
-                                      (remove-if-not
-                                       (lambda (s) (string= (symbol-name s) sym-name))
-                                       (coalton-impl/algorithm::immutable-map-keys
-                                        (tc:environment-value-environment (tc-env-env env)))))))
-                       (error 'tc-error
-                              :err (coalton-error
-                                    :span (parser:node-source var)
-                                    :file file
-                                    :message "Unknown variable"
-                                    :primary-note "unknown variable"
-                                    :help-notes (mapcar
-                                                 (lambda (symbol)
-                                                   (error:make-coalton-error-help
-                                                    :span (parser:node-source var)
-                                                    :replacement (lambda (s)
-                                                                   (declare (ignore s))
-                                                                   (format nil "~S" symbol))
-                                                    :message (format nil "Did you mean ~S?" symbol)))
-                                                 matches))))))
-
-         (qual-ty (tc:fresh-inst scheme))
-
-         (ty (tc:qualified-ty-type qual-ty))
-
-         (preds (tc:qualified-ty-predicates qual-ty)))
-
-    (values
-     ty
-     (loop :for pred :in preds
-           :collect (tc:make-ty-predicate
-                     :class (tc:ty-predicate-class pred)
-                     :types (tc:ty-predicate-types pred)
-                     :source (parser:node-source var))))))
+  (let* ((var-name (parser:node-variable-name var))
+         (scheme (or (lookup-var-type env var-name)
+                     (lookup-var env var-name))))
+    (unless scheme
+      ;; Variable is unbound: create an error
+      (error 'tc-error
+             :err (coalton-error
+                   :span (parser:node-source var)
+                   :file file
+                   :message "Unknown variable"
+                   :primary-note "unknown variable"
+                   :help-notes (mapcar
```

Comment:
there's a mapcar here and a loop-collect below. can we just make this a loop-collect as well?